src/metabase/models/card.clj

```diff
@@ -237,6 +239,49 @@
       (field-values/update-field-values-for-on-demand-dbs! field-ids))
     (create-actions-when-is-writable! card)))
 
+(defn- assoc-viz-setting
+  [card key val]
+  (assoc-in card [:visualization_settings key] val))
+
+(defmulti ^:private migrate-viz-settings*
+  "Migration visualization_settings to the appropriate version"
+  {:arglists '([card desired-version])}
+  (fn [card desired-version]
+    (let [current-version (or (get-in card [:visualization_settings :version])
+                              1)]
+      (if (= current-version desired-version)
+        ::identity
+        [current-version desired-version]))))
+
+(defmethod ^:private migrate-viz-settings* ::identity [card _]
+  card)
+
+(defmethod ^:private migrate-viz-settings* [1 2] [{:keys [visualization_settings] :as card} _]
```

Comment:
I suppose the idea is we can add new 'migrations' by adding a new method `[2 3]`, `[3 4]`, etc.

Then, when we call the migration, if the desired version is a few steps away, eg `[1 4]`, we can just chain the migrations with something like 

`(map (partial migrate-viz-settings* card) (partition 2 1 (range current-version desired-version)`

Am I thinking about this correctly?

Additionally, does it make sense to have a `:default` method that either throws or just passes the viz-settings along, but warns or something?

Reply:
> Am I thinking about this correctly?

yep!

> Additionally, does it make sense to have a :default method that either throws or just passes the viz-settings along, but warns or something?

I think the only potential error is if (for whatever reason) there's a `viz_settings->version` in the database that's not falsey, `1`, or `2`. In that case we'll get a method missing error with whatever the version actually was (e.g., `3`), which feels like good behavior to me